workflows/cloudify_system_workflows/snapshot.py

```diff
@@ -0,0 +1,574 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+
+import json
+import tempfile
+import shutil
+import zipfile
+import itertools
+import os
+import subprocess
+
+import elasticsearch
+import elasticsearch.helpers
+
+from datetime import datetime
+
+from cloudify import utils
+from cloudify.context import BootstrapContext
+from cloudify.decorators import workflow
+from cloudify.exceptions import NonRecoverableError
+from cloudify.manager import get_rest_client
+from cloudify_system_workflows.deployment_environment import \
+    generate_create_dep_tasks_graph
+
+
+_METADATA_FILE = 'metadata.json'
+
+# metadata fields
+_M_HAS_CLOUDIFY_EVENTS = 'has_cloudify_events'
+_M_VERSION = 'snapshot_version'
+
+_VERSION = '3.3'
```

Comment:
The version should be retrieved by using REST API's version call.
Note that the version includes milestone information as well (don't know if relevant here).